server/src/main/java/org/candlepin/messaging/impl/artemis/ArtemisContextListener.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.messaging.impl.artemis;
+
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.common.config.Configuration;
+import org.candlepin.messaging.CPMContextListener;
+import org.candlepin.messaging.CPMException;
+
+import org.apache.activemq.artemis.core.server.embedded.EmbeddedActiveMQ;
+
+import com.google.inject.Injector;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+import javax.inject.Singleton;
+
+
+
+/**
+ * CPMContextListener implementation backed by Artemis
+ */
+@Singleton
+public class ArtemisContextListener implements CPMContextListener {
+    private static  Logger log = LoggerFactory.getLogger(ArtemisContextListener.class);
+
+    private Configuration config;
+    private EmbeddedActiveMQ activeMQServer;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize(Injector injector) throws CPMException {
+        this.config = injector.getInstance(Configuration.class);
+
+        // Create the server if we're configured to do so
+        boolean embedded = this.config.getBoolean(ConfigProperties.ACTIVEMQ_EMBEDDED);
+
+        if (embedded) {
```

Comment:
I don't feel this is the correct place for such a message. I've, instead, added the following log entry in the session factory:

```
log.info("Connecting to Artemis server at {}", brokerUrl);
```

This will indicate exactly where the server lives, in both the embedded and remote cases.